File: python-backend/src/app/internal_dto/internal_interview_round_detail_response.py
from pydantic import BaseModel, Field


class InterviewOptionResponse(BaseModel):
    key: str = Field(..., alias="key")
    text: str = Field(..., alias="text")


class InternalInterviewRoundDetailResponse(BaseModel):
    round_id: int = Field(..., alias="roundId")
    run_id: int = Field(..., alias="runId")
    round_no: int = Field(..., alias="roundNo")
    question_text: str = Field(..., alias="questionText")
    options: list[InterviewOptionResponse] = Field(default_factory=list)
    user_answer: str | None = Field(default=None, alias="userAnswer")
    status: str | None = Field(default=None)
    analysis_json: str | None = Field(default=None, alias="analysisJson")

File: java-backend/common/src/main/java/com/elias/common/dto/interview/internal/response/InternalInterviewRoundDetailResponse.java
package com.elias.common.dto.interview.internal.response;

import com.elias.common.dto.interview.response.InterviewOptionResponse;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * 面试题目答案详情响应
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class InternalInterviewRoundDetailResponse {

    /**
     * 题目记录ID
     */
    private Long roundId;

    /**
     * run ID
     */
    private Long runId;

    /**
     * 轮次编号
     */
    private Integer roundNo;

    /**
     * 题干
     */
    private String questionText;

    /**
     * 选项列表
     */
    private List<InterviewOptionResponse> options;

    /**
     * 用户回答
     */
    private String userAnswer;

    /**
     * 分析结果 JSON
     */
    private String analysisJson;

    /**
     * 状态
     */
    private String status;
}